services/blockchain-indexer/shared/dataService/business/events.js

```diff
@@ -119,24 +120,14 @@ const getEvents = async (params) => {
 		queryParams = normalizeRangeParam(queryParams, 'timestamp');
 	}
 
-	if (params.topic) {
-		const { topic, ...remQueryParams } = queryParams;
-		queryParams = remQueryParams;
-
-		queryParams.whereIn = {
-			property: 'topic',
-			values: topic.split(','),
-		};
-	}
-
 	if (params.transactionID) {
 		const { transactionID, ...remQueryParams } = queryParams;
 		queryParams = remQueryParams;
 
 		if (!params.topic) {
-			queryParams.topic = transactionID;
+			params.topic = transactionID;
```

Comment:
The reason for cloning `params` in top with `queryParams` is not to modify the user supplied params. should it be `queryParams`? if so, let's change other occurrences as well. 

```suggestion
			queryParams.topic = transactionID;
```

Reply:
This is because of a slight change in approach. I reverted to our old approach where we first fetch all the event IDs which include all topics specified and then query the eventsTable using those topicIDs instead of using joins.
https://github.com/LiskHQ/lisk-service/blob/e5c4d988756f51b322dc73d4f95f2234d90df1b9/services/blockchain-indexer/shared/dataService/business/events.js#L118

params do not correctly depict the way we store data in db. Since we store transactionID and senderAddress as topics, I have modified params to include all 3 cases in a comma separated format.